### PR TITLE
Added hyperparameter optimization support

### PR DESCRIPTION
**Pull request by harshgupta7**

### CODE REVIEW

**Comment by maciejkula** [comment]
I think making hyperparameter optimization for LightFM is a great idea!

I would love to have an example that goes through the process of hyperparameter search, maybe using a couple of different approaches and libraries. However, because there are different ways of optimizing hyperparameters, I think it may be best to keep the specifics separate from the core of the library, and let the users figure out how they want to do it.

Would you be open to converting this to an example?

**Comment by harshgupta7** [comment]
Sure! I'll put it in the doc.

**Comment by maciejkula** [comment]
Great! Maybe a Jupyter notebook we could include along with the existing
examples?

**Comment by harshgupta7** [comment]
Sounds good..will do!

File: lightfm/lightfm.py [hunk 1]
```diff
@@ -926,3 +929,185 @@ def set_params(self, **params):
...
+                train[user, test_interactions] = 0.
+                test[user, test_interactions] = interactions[user, test_interactions]
+
+            return train.tocsr(), test.tocsr()
+
+        train_interactions, test_interactions = train_test_split(interactions=interactions,\
+            split_count=split_count)
+
+
+        def obective(params):
```
Comment (RAbraham) [inline lightfm/lightfm.py]: typo `objective`?

Comment (harshgupta7) [reply lightfm/lightfm.py]: Yeah. Will fix.

File: lightfm/lightfm.py [hunk 2]
```diff
@@ -926,3 +929,185 @@ def set_params(self, **params):
...
+
+        num_threads: int, optional
+            Number of parallel computation threads to use. Should
+            not be higher than the number of physical cores.
+
+        evaluation: string, optional
+            evaluation method to be used for objective function
+            one of ('auc', 'precision_at_k')
+
+        split: int, optional
```
Comment (RAbraham) [inline lightfm/lightfm.py]: minor change: `split` to `split_count`

**Comment by maciejkula** [comment]
Hey @harshgupta7 --- I'd still love an example!

I'm going to close this for now, but let me know if you'd like a hand with the example.

**Pull request closed without merging**